### ISSUE TITLE
testNative tag ignores binary name generated by using quarkus.package.output-name/add-runner-suffix

### ISSUE THREAD

**Issue by rivasdiaz** [quarkusio/quarkus#30101]
### Describe the bug

If the properties `quarkus.package.output-name=<binary-name>` or `quarkus.package.add-runner-suffix=false` are used to modify the binary name generated by the `quarkusBuild`, the `testNative` task fails to find the executable to run.

Once both properties are removed from the configuration, the `testNative` task works again.

### Expected behavior

The `testNative` task correctly finds the native binary with the custom name and is able to run the native tests.

### Actual behavior

The `testNative` tasks fails. After opening the generated test results report, the following exception is listed: 

```
java.lang.RuntimeException: java.io.IOException: Cannot run program "/private/tmp/test/build/test-1.0.0-SNAPSHOT-runner": error=2, No such file or directory
	at io.quarkus.test.junit.QuarkusIntegrationTestExtension.throwBootFailureException(QuarkusIntegrationTestExtension.java:337)
	at io.quarkus.test.junit.QuarkusIntegrationTestExtension.beforeEach(QuarkusIntegrationTestExtension.java:110)
        [...]
```

### How to Reproduce?

Steps to reproduce:
1. Create a simple quarkus application. The default example comes with a native test: `quarkus create app --gradle test`.
2. Verify native tests work by default by running: `./gradlew -Dquarkus.package.type=native build testNative`.
3. Make sure to clean the previous build so the currently generated binary is deleted, by running: `./gradlew clean`.
4. Use any of the two properties `quarkus.package.output-name` or `quarkus.package.add-runner-suffix` while building: `./gradlew -Dquarkus.package.type=native -Dquarkus.package.output-name=test -Dquarkus.package.add-runner-suffix=false build testNative`.
5. Open the generated native test report to see the exception: `file:///private/tmp/test/build/reports/tests/testNative/index.html`

### Output of `uname -a` or `ver`

Darwin rrivas-mac 22.1.0 Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:54 PDT 2022; root:xnu-8792.41.9~2/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "17.0.5" 2022-10-18 OpenJDK Runtime Environment GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08) OpenJDK 64-Bit Server VM GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08, mixed mode, sharing)

### Quarkus version or git rev

2.15.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.5.1

### Additional information

Although I chose `gradle` for the reproduction steps, it can be reproduced with both `gradle` and `maven`.

**Comment by geoand** [comment]
@jacobdotcosta would you be interested in looking into this one?

**Comment by jacobdotcosta** [comment]
Yes, I'll take a look. Thanks.

**Comment by geoand** [comment]
👌

**Comment by rivasdiaz** [comment]
This issue is fixed for me in 2.16.0.CR1 in both Maven and Gradle.
Thanks!
